### PR TITLE
Feature: Installed node modules / Additional Info

### PR DESCRIPTION
**Pull request by frankebersoll**
Somehow, I managed to corrupt the [other pull request](https://github.com/exceptionless/Exceptionless.JavaScript/pull/33), so here is a new one:

This feature (Issue #6) adds a `NodeModuleCollector.ts` that tries to find all currently loaded npm modules:
1. It synchronously runs `npm ls --depth=0 --json` to get all installed packages on root level. Those modules are cached, as they won't change during runtime.
2. It checks `require.cache` to get all currently cached modules. Those are all JavaScript files that have been required since startup of the application.
3. It crops all those paths to get the portion behind `/node_modules/`, which is the module name, and makes them unique.
4. It returns all installed packages from 1. that match those module names.

Furthermore, it iterates all exception properties and puts them in the `@ext` data field on the error object, showing them as "Additional Data" in the UI (Issue #8). The following properties are currently ignored: 

```
'arguments',
'column',
'columnNumber',
'description',
'fileName',
'message',
'name',
'number',
'line',
'lineNumber',
'opera#sourceloc',
'sourceURL',
'stack',
'stacktrace'
```

### CODE REVIEW

File: src/plugins/default/ErrorPlugin.ts [hunk 1]
```diff
@@ -27,4 +52,24 @@ export class ErrorPlugin implements IEventPlugin {
 
     next && next();
   }
+
+  private getAdditionalData(exception: Error): { [key: string]: any } {
+    let keys = Object.keys(exception)
+      .filter(key => this.ignoredProperties.indexOf(key) < 0);
```
Comment (niemyjski) [inline src/plugins/default/ErrorPlugin.ts]: Might want to check to see if the property is a function here.

File: src/plugins/default/ErrorPlugin.ts [hunk 2]
```diff
@@ -20,6 +37,14 @@ export class ErrorPlugin implements IEventPlugin {
 
         let result = parser.parse(context, exception);
         if (!!result) {
+          let additionalData = this.getAdditionalData(exception);
+          if (!!additionalData) {
```
Comment (niemyjski) [inline src/plugins/default/ErrorPlugin.ts:43]: If additional data is an empty object it shouldn't be added to extra properties as we will render empty extra exception properties if the key exists.

**Comment by niemyjski** [comment]
Looks great! I just noticed a use case where the extended data item could be added if any extra properties were functions (which would then return an empty object). If you want I can make this change.

**Comment by frankebersoll** [comment]
@niemyjski Thanks for your review; just made some enhancements and added more unit tests, that should do the trick. Empty extended data won't be added now.

**Comment by niemyjski** [comment]
Great job! I was thinking about this more this morning and I think that once we get our changes to parse depth and exclusions be generic.. We could come back to this and use that for parsing additional properties (share more code).

**Pull request merged**